src/Transport/Sending/IBatcher.cs

```diff
@@ -1,10 +1,11 @@
 namespace NServiceBus.Transport.AzureServiceBus
 {
     using System.Collections.Generic;
+    using System.Threading.Tasks;
     using Transport;
 
     interface IBatcherInternal
     {
-        IList<BatchInternal> ToBatches(TransportOperations operations);
+        Task<List<BatchInternal>> ToBatches(TransportOperations operations);
```

Comment:
Rather than doing
```c#
namespaceBundleConfigurations.Value.GetAwaiter().GetResult();
```

I'd suggest to perform a blocking call in the Lazy block, replacing `AsynLazy` with `Lazy`.